### DISCUSSION TITLE
Does Seaweedfs version upgrade requires migration?

### DISCUSSION THREAD

**Discussion by immad-imtiaz** [seaweedfs/seaweedfs#2399] [General]
I have a docker compose file where I am replace SeaweedFs with MinIO, I run the script in a single container as below, which show the `docker-entrypoint.sh` we have a migration script as well that runs after all the required process are up and running. However,

1. I would like to know if in future I upgrade the seaweedfs version would this also require a migration.
Given /data2 is docker volume provided in docker compose file. 

2. Also please recommend if there is a better way of running the seaweedfs than my script.

3. How can I configure logs for each process and if there is any log rotation mechanism built in that I can use

```
#!/bin/sh
# Enable erasure coding for warm storage 
# https://github.com/chrislusf/seaweedfs/wiki/Erasure-Coding-for-warm-storage
LOG_LOCATION=/opt/output/logs/seaweedfs
mkdir -p $LOG_LOCATION

/usr/bin/weed scaffold -config=master > /etc/seaweedfs/master.toml

# Start a master server
/usr/bin/weed master -mdir="/data2/master" -ip="127.0.0.1" -volumeSizeLimitMB=1024 -port=9333 > $LOG_LOCATION/master.log 2>&1 &
# Start a volume server
/usr/bin/weed volume -dir="/data2/volume" -index=leveldb -max=1024 -mserver="127.0.0.1:9333" -port=8080 -dataCenter=dc1 -rack=rack1 > $LOG_LOCATION/volume-server-1.log 2>&1 &
# Start a filer server
/usr/bin/weed filer -port=8888 -dirListLimit=100000 -master="127.0.0.1:9333" > $LOG_LOCATION/filer.log 2>&1 &
# First build the credentials config to be used by s3 server
/usr/bin/build-configs.sh > $LOG_LOCATION/build-config.log 2>&1
# Start the S3 server
/usr/bin/weed s3 -port=9000 -config=/usr/bin/s3-credentials-config.json > $LOG_LOCATION/s3-gateway.log 2>&1 &
# Run the migration script
/usr/bin/data-migration.sh > $LOG_LOCATION/data-migration.log 2>&1 &
# Keep the container running
tail -f /dev/null
```

**Comment by chrislusf** [comment]
1. Most (if not all) upgrades would not need any data migration.
2. You can use one process `weed server -s3` directly to simplify the steps. The options can be read from a file https://github.com/chrislusf/seaweedfs/wiki/Load-Command-Line-Options-from-a-file
3. The logs is using glog. The log files are already in `/tmp` directory.
